curator-framework/src/main/java/org/apache/curator/framework/CuratorFramework.java

```diff
@@ -331,4 +329,29 @@
      * @return true/false
      */
     boolean isZk34CompatibilityMode();
+
+    /**
+     * Calls {@link #notifyAll()} on the given object after first synchronizing on it. This is
+     * done from the {@link #runSafe(Runnable)} thread.
+     *
+     * @param monitorHolder object to sync on and notify
+     * @since 4.1.0
+     */
+    default void postSafeNotify(Object monitorHolder)
```

Comment:
Thinking about the future...
Would it be better to make it return a CompletableFuture?
This way the caller will have a chance to know when eventually the operation has been executed

Reply:
Good idea - I'll add that